nav2-usability/scenario_code/lib/nested-routing/nested_routing_auto_route.dart

```diff
@@ -0,0 +1,181 @@
+// Copyright 2021, the Flutter project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// Nested example
+/// Done using AutoRoute
+import 'package:auto_route/auto_route.dart';
+
+import 'package:flutter/material.dart';
+import 'nested_routing_auto_route.gr.dart';
+
+void main() {
+  runApp(BooksApp());
+}
+
+// Declare routing setup
+@MaterialAutoRouter(
+  replaceInRouteName: 'Screen,Route',
+  routes: <AutoRoute>[
+    AutoRoute(
+      page: AppScreen,
+      path: "/",
+      children: [
+        RedirectRoute(path: "", redirectTo: "books/new"),
+        AutoRoute(path: 'books/:tab', page: BooksScreen),
+        AutoRoute(path: 'settings', page: SettingsScreen),
+      ],
+    ),
+    RedirectRoute(path: "*", redirectTo: "/")
+  ],
+)
+class $AppRouter {}
+
+class BooksApp extends StatelessWidget {
+  final _appRouter = AppRouter();
+
+  @override
+  Widget build(BuildContext context) {
+    return MaterialApp.router(
+      routerDelegate: _appRouter.delegate(),
+      routeInformationParser:
+          _appRouter.defaultRouteParser(includePrefixMatches: true),
+    );
+  }
+}
+
+class AppScreen extends StatelessWidget {
+  @override
+  Widget build(BuildContext context) {
+    return AutoTabsScaffold(
```

Comment:
I am amazed (in a good way) that auto route provides these built-in widgets. However, the scenario meant to showcase how user can write arbitrary widget that enclose  BooksRoute, SettingsRoute. Is it possible to write your own Scaffold without using the built-in one?

Reply:
Hi @chunhtai , is this acceptable? It's using Scaffold now

Reply:
@chunhtai sure thing, the scaffold is just to avoid the not so pretty approach when you have to build the scaffold inside of the AutoTabsRouter's body to access the scoped controller or use a global key.
Users can also **Optionally** use a simplified declarative approach which I believe will look more familiar but the catch is that they have to handle tab switching on url changes in `onNavigate`
```dart
     Scaffold(
        body: AutoTabsRouter.declarative(
          activeIndex: activeIndex,
           routes: const [
             BooksRoute(),
             SettingsRoute(),
           ],
          onNavigate: (match, initial) {
            activeIndex = (match.routeName == BooksRoute.name) ? 0 : 1;
            if (!initial) {
              setState(() {});
            }
          },
          
        ),
```